forum/frontend/src/components/register.js

```diff
@@ -28,109 +28,135 @@ const Register = () => {
     return null;
   };
 
-  const handleSubmit = async (e) => {
-    e.preventDefault();
-    
-    const validationErrorMessage = validateInputs();
-    if (validationErrorMessage) {
-      setValidationError(validationErrorMessage);
+  const handleSubmitForm = (username, email, password, firstName, lastName, phone) => {
+    setLoading(true);
+    setError('');
+    setSuccess('');
+
+    const validationError = validateInputs();
+    if (validationError) {
+      setError(validationError);
+      setLoading(false);
       return;
     }
 
-    setLoading(true);
-    
-    try {
-      const response = await axios.post('http://localhost:8000/api/v1/register/', {
-        email: email,
-        password: password,
-        username: username,
-        first_name: firstName,
-        last_name: lastName,
-        phone: phone,
-      });
-
-      console.log('User registered successfully:', response.data);
-      navigate('/login');
-    } catch (error) {
-      if (error.response) {
-        const { data } = error.response;
-        if (data.email) {
-          setError('This email is already in use.');
-        } else if (data.username) {
-          setError('This username is already taken.');
-        } else {
-          setError('Registration failed.');
-        }
-      } else {
-        setError('An error occurred. Please try again.');
+    api.post('/api/v1/register/', {
+      username,
+      email,
+      password,
+      first_name: firstName,
+      last_name: lastName,
+      phone,
```

Comment:
Add validation for phone to match the expected format and ensure password meets complexity requirements.